Program.cs

```diff
@@ -25,47 +47,61 @@ static void Main(string[] args)
                         AddCharacter();
                         break;
                     case 2:
-                        AddParty();
+                        ShowCharacters();
                         break;
                     case 3:
-                        AddEncounter();
+                        AddParty();
                         break;
                     case 4:
-                        StartEncounter();
+                        AddEncounter();
                         break;
                     case 5:
+                        StartEncounter();
+                        break;
+                    case 6:
                         Quit();
                         break;
                     default:
                         break;
                 }
             }
-          
         }
 
-
         static void AddCharacter()
         {
-            Menu CharacterMenu = new Menu("Choose an Option", new string[] { "Enter Name", "Enter HP", "Enter Damage" });
+            bool goBack = false;
```

Comment:
I moved the loop control variable inside the functions where they are used. This helps prevent them from being contaminated accidentally by other functions.